### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by Kazhuu**
Remove Python 2 and support for Python 3 only. Python 2 is deprecated and not recommended to use anymore. I used to following scripts to test the build
```
make -j4 && tools/scripts/compiletest.sh -cq
tce-selftest -v
```

I tried to look for Python files what I could find. Some of them might not be used by the above testing scripts. Feel free to point out if I missed something. On many files I fixed formatting issues, spaces and tabs mixed to only use spaces and updated python 2 functions to use python 3 instead. Also found some already existing issues like misspelled variable names.

Comments welcome!

### CODE REVIEW

File: tce/Python-bindings/scripts/cfgTest.py [hunk 1]
```diff
@@ -39,34 +39,34 @@ def printOut(pr):
...
-        blocks = blocks + stats.normalBBCount()        
-    print "Program:"+sys.argv[1]
-    print "\tMoves:",moveSum  
-    print "\tImmediates:",immediateSum 
-    print "\tInstructions:",instructionSum 
-    print "\tBasicBlocks:",blocks
-    print "\tProcedures:",pr.procedureCount()
-    print "\tMovesPerCycle:",1.0*moveSum/instructionSum 
+        blocks = blocks + stats.normalBBCount()
+    print("Program:"+sys.argv[1])
```
Comment (pjaaskel) [inline tce/Python-bindings/scripts/cfgTest.py:43]: The style looks tight. Can you add white spaces around the + operator?

**Review by pjaaskel — CHANGES_REQUESTED** [review]
Most of this patch is not about the python3 upgrade but white space / style changes. Usually I have not allowed large style / WS commits since they make utilizing 'git blame' more complex when tracking reason commits (and authors) for code (logic) changes. How much additional work would it be to drop those style changes from this commit to make this only about the necessary updates for the python2 to python3 transition?

File: tce/Python-bindings/scripts/cfgTest.py (hunk 1 above)
Comment (Kazhuu) [reply tce/Python-bindings/scripts/cfgTest.py:43]: Okay if you are really want and care about git blame over these improvements I can make it. I guess if you follow this mentality then upgrading code formatting cannot basically be done? I can remove the changes of course if that's what is needed :) I personally care about avoiding unnecessary white spaces and follow PEP8 formatting. And usually fix some on the same way when improving and fixing the code base. Leaving things "cleaner" than before on the process. Mostly those edits are coming from my used formatters from my editor. Of course the code is full of format errors and this change just touches the surface.

Comment (pjaaskel) [reply tce/Python-bindings/scripts/cfgTest.py:43]: Overall, maintaining easy 'git blame' is extremely valueable in a long running project like TCE with a lot of developers that come and go. But in this particular case I don't see so much value so I'll just pull this in. But in the future, style updates are preferably done only nearby the actual changed logic and in a separate commit to make log readability better (bunch of style updates can bury the interesting changes in the noise).

**Pull request merged**